src/main/js/controls/Bubble/helpers/colorGradient.js

```diff
@@ -44,3 +44,52 @@ export const generateColor = (dataTarget) => {
 
     return huePaletteList;
 };
+
+/**
+ * Generates color range for the given color for a single dataset
+ *
+ * @private
+ * @param {object} dataTarget - data for the bubble graph
+ * @returns {d3.scale} - returns function to get color for specific bubble
+ */
+export const generateColorSingleDataset = (dataTarget) => {
+    let radiusData;
+    let color = [];
+    let huePaletteList = null;
+
+    if (utils.isUndefined(dataTarget.weight)) {
+        radiusData = dataTarget.values.map((element) => element.y);
+    } else {
+        radiusData = dataTarget.values.map((element) =>
+            bubbleScale(dataTarget)(element.weight)
+        );
+    }
+
+    if (utils.isDefined(dataTarget.palette)) {
+        color = dataTarget.palette;
```

Comment:
I asked this above but what happens if a consumer provides a color palette that is invalid like `BUBBLE.PALETTE.OARNGE` instead of `BUBBLE.PALETTE.ORANGE`?

Reply:
In that case the bubbles will be rendered as gray.